cli/lib/kontena/cli/stacks/remove_command.rb

```diff
@@ -10,12 +10,46 @@ class RemoveCommand < Kontena::Command
 
     parameter "NAME", "Stack name"
     option "--force", :flag, "Force remove", default: false, attribute_name: :forced
+    option "--keep-dependent", :flag, "Do not remove dependencies"
+    option "--ignore-not-found", :flag, "Ignore stack not found errors", hidden: true
 
     requires_current_master
     requires_current_master_token
 
+    def fetch_stack
+      client.get("stacks/#{current_grid}/#{name}")
+    rescue Kontena::Errors::StandardError => ex
+      if ex.status == 404 && ignore_not_found?
+        puts "#{pastel.yellow('Warning:')} The stack #{pastel.cyan(name)} does not exist."
+        exit 0
+      end
+      raise ex
+    end
+
+    def confirm_remove(stack)
+      if stack['parent']
+        puts "#{pastel.yellow('Warning:')} The stack #{pastel.cyan(stack['parent']['name'])} depends on stack #{name}"
+      end
+      if stack['children'] && !stack['children'].empty?
+        puts "#{pastel.yellow('Warning:')} The stack #{pastel.cyan(name)} has dependencies that will be removed:"
+        stack['children'].each do |child|
+          puts "- #{pastel.yellow(child['name'])}"
+        end
+      end
+      confirm_command(name)
+    end
+
     def execute
-      confirm_command(name) unless forced?
+      stack = fetch_stack
+      confirm_remove(stack) unless forced?
+      stack.fetch('children', Hash.new).each do |child_stack|
+        caret"Removing dependency #{pastel.cyan(child_stack['name'])}"
+        cmd = ['stack', 'remove', '--ignore-not-found']
```

Comment:
The `{"children": [ ... ]}` array returned by the server API will only include stacks that actually exist?

Is this `--ignore-not-found` really even needed? That would imply some bug where the same stack shows up twice in the dependency tree?

Reply:
In case you have force removed something from between or your stack-tree failed to install completely. The `--ignore-not-found` will remove everything and not die even if a stack from the dependency tree is missing for some reason.

Reply:
But yes, the children array is built from stacks that actually exist so the flag might actually be of no use.